add steps to migrate the projects to use the sdk version 0.12 (#2185)
operator-framework/operator-sdk: doc/migration/version-upgrade-guide.md

  deploy/olm-catalog/
  └── memcached-operator
      ├── 0.1.0

>       │   └── memcached-operator.v0.1.0.clusterserviceversion.yaml

Author: ```suggestion
      │   └── memcached-operator.v0.1.0.clusterserviceversion.yaml
```

      ├── 0.2.0

>       │   └── memcached-operator.v0.2.0.clusterserviceversion.yaml

Reviewer: Nit: This change seems inadvertant/unrelated? Looking at the file with hexdump, it looks like `master` has two non-breaking spaces (`a0 20`) and your branch has two breaking spaces (`20`). Not sure if that matters, but since this is unrelated, I'd suggested reverting these lines.

>       │   └── memcached-operator.v0.2.0.clusterserviceversion.yaml

Author: ```suggestion
      │   └── memcached-operator.v0.2.0.clusterserviceversion.yaml
```

    - Delete the old CRD manifest files.
    - Rename CR manifest file names from `<group>_<version>_<kind>_cr.yaml` to `<full group>_<version>_<kind>_cr.yaml`.

## `v0.12.x`

### Go version

- Ensure that you are using a go version 1.13+

> ### modules

Reviewer: We need more detail about dependencies because we no longer support `dep`/`Gopkg.toml`. We should tell users how to [migrate to Go modules](https://blog.golang.org/migrating-to-go-modules) if they are still using `dep`.

> - Replace your `go.mod` file for:

Reviewer: It doesn't look like we have been telling users to do a full replacement of their dependency files, and for good reason, I think. Users may have made their own custom changes to bring in more dependencies. This instruction could cause their custom dependency inclusions to break or change in unexpected ways.

Rather, we should document which pieces of our `go.mod` scaffold changed between `v0.11.x` and `v0.12.x`.

	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.0.0-20190918202959-c340507a5d48
	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20190612205613-18da4a14b22b
	k8s.io/component-base => k8s.io/component-base v0.0.0-20190918200425-ed2f0867c778
	k8s.io/cri-api => k8s.io/cri-api v0.0.0-20190817025403-3ae76f584e79
	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20190918203248-97c07dcbb623
	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20190918201136-c3a845f1fbb2
	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20190918202837-c54ce30c680e
	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20190918202429-08c8357f8e2d
	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20190918202713-c34a54b3ec8e
	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20190918202550-958285cf3eef
	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20190918203421-225f0541b3ea
	k8s.io/metrics => k8s.io/metrics v0.0.0-20190918202012-3c1ca76f5bda
	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190918201353-5cc279503896
)
```
- Run `go mod tidy` to update the project modules

> - Run the command `operator-sdk generate k8s` to ensure that your resources will be updatedd

Reviewer 2: typo: updatedd should be updated

> - Run the command `operator-sdk generate k8s` to ensure that your resources will be updatedd

Author: ```suggestion
- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
```

- Run the command `operator-sdk generate openapi` and ensure that no errors such as `API rule violation` are raised. For further information see the [API rules][api-rules] documentation. 

> ### Main Go updates (Optional)

Reviewer: ```suggestion
### (Optional) Update your operator to print its version
```

> The following change is to get the feature : Added `Operator Version: X.Y.Z` information in the operator logs.([#1953](https://github.com/operator-framework/operator-sdk/pull/1953))

Reviewer: ```suggestion
In v0.12.0, the SDK team updated the scaffold for `cmd/manager/main.go` to include the operator's version in the output produced by the `printVersion()` function. See [#1953](https://github.com/operator-framework/operator-sdk/pull/1953)
```

> In the `<project>/cmd/manager/main.go` file, add the follow lines.

Reviewer: ```suggestion
To add this feature to your operator, add the following lines in `<project>/cmd/manager/main.go`:
```

```go
    ...
    "github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/apis"
	"github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/controller"
	"github.com/operator-framework/operator-sdk-samples/memcached-operator/version" //This line
    ...
    func printVersion() {
        log.Info(fmt.Sprintf("Operator Version: %s", version.Version)) //This line
        log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))

> ```

Reviewer: This snippet is confusing, I think, and spacing is a bit off.

How about the following (and note that I'm using tabs to conform to `go fmt`):
```go
import (
	...
	"<your_module_path>/version"
	...
)

func printVersion() {
	log.Info(fmt.Sprintf("Operator Version: %s", version.Version))
	...
}
```

        k8s.io/cri-api => k8s.io/cri-api v0.0.0-20190817025403-3ae76f584e79
        k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20190918203248-97c07dcbb623
        k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20190918201136-c3a845f1fbb2
        k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20190918202837-c54ce30c680e
        k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20190918202429-08c8357f8e2d
        k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20190918202713-c34a54b3ec8e
        k8s.io/kubelet => k8s.io/kubelet v0.0.0-20190918202550-958285cf3eef
        k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20190918203421-225f0541b3ea
        k8s.io/metrics => k8s.io/metrics v0.0.0-20190918202012-3c1ca76f5bda
        k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190918201353-5cc279503896
    )
    ...
```

**NOTE**: Check [here](https://github.com/operator-framework/operator-sdk-samples/pull/90/files#diff-e15cac8b95d260726ca9db9fb25d9230) an example of this upgrade to see the changes from the version `0.11.0` to `0.12.0`.

- Run `go mod tidy` to update the project modules

> - Run the command `operator-sdk generate k8s` to ensure that your resources will be updatedd

Author: ```suggestion
- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
```
